### PR TITLE
fix: concealer breakage on carryover tag set when calculating todo items

### PR DESCRIPTION
**Pull request by metalelf0**
This PR actually fixes issue #432 by using recursion to iterate over all descendant nodes and counting all todo_items.

### CODE REVIEW

File: lua/neorg/modules/core/norg/concealer/module.lua [hunk 1]
```diff
@@ -66,6 +66,27 @@ local function schedule(func)
...
+            local todo_item_type = type_node:type():sub(string.len("todo_item_") + 1)
+            local resulting_todo_item = results[todo_item_type] or 0
+
+            results[todo_item_type] = resulting_todo_item + 1
+            results.total = results.total + (todo_item_type == "cancelled" and 0 or 1)
+        end
+    end
+end
+
+local function traverse_nodes(root_node, results, func)
```
Comment (max397574) [inline lua/neorg/modules/core/norg/concealer/module.lua:83]: why the `func` parameter
this function is only used here and always the same function is used

File: lua/neorg/modules/core/norg/concealer/module.lua [hunk 2]
```diff
@@ -66,6 +66,27 @@ local function schedule(func)
     end)
 end
 
+local function add_to_counters_if_todo_node(node, results)
+    if vim.startswith(node:type(), "todo_item") then
+        local type_node = node:named_child(1)
+
+        if type_node then
+            local todo_item_type = type_node:type():sub(string.len("todo_item_") + 1)
```
Comment (max397574) [inline lua/neorg/modules/core/norg/concealer/module.lua:74]: seems really weird to do that like this
just use the number or  (cleaner imo) a pattern

File: lua/neorg/modules/core/norg/concealer/module.lua (hunk 1 above)
Comment (metalelf0) [reply lua/neorg/modules/core/norg/concealer/module.lua:83]: I passed `func` as a parameter to make it more extendable. You can pass a custom function and have the code count items depending on different rules, or do different stuff on nodes. I get your point though, so if you prefer to keep it simple I can remove the func parameter and make it use the `add_to_counters_if_todo_node` function directly.

File: lua/neorg/modules/core/norg/concealer/module.lua (hunk 2 above)
Comment (metalelf0) [reply lua/neorg/modules/core/norg/concealer/module.lua:74]: I just copy-pasted the original implementation. Not sure why it was done this way, maybe @vhyrro can tell us?

File: lua/neorg/modules/core/norg/concealer/module.lua (hunk 1 above)
Comment (max397574) [reply lua/neorg/modules/core/norg/concealer/module.lua:83]: I think you could
we can later change this if we want
but I see no reason why we'd do this

File: lua/neorg/modules/core/norg/concealer/module.lua (hunk 2 above)
Comment (vhyrro) [reply lua/neorg/modules/core/norg/concealer/module.lua:74]: Magic numbers in my opinion are way worse than less readable code - using a pattern is overkill to me (why an entire regex system to cut a string?). I'm pretty annoyed that `len()`'s return value is incompatible with `:sub()` and requires a `+ 1`, but it's lua what can ya do :(

**Comment by vhyrro** [comment]
Looks good to me! Code is nice, can't seem to find any bugs with it - thanks a bunch for the PR!

**Pull request merged**